Docs/update concepts (#228)
wasmCloud/wasmCloud.com: docs/concepts/capabilities.mdx

For example, when writing a web service that exposes a [RESTful API][restful-api], the HTTP server and the database are not parts of your business logic -- your primary concern is structuring your API so that users can easily get the data they need.

[restful-api]: https://en.wikipedia.org/wiki/REST

### Capability Contracts

:::info

> The following section was written before WASI Preview 2 was stable. wasmCloud supports the [wasi-cloud-core](https://github.com/WebAssembly/wasi-cloud-core) interfaces, the existing `wasmcloud:*` interfaces, and custom interfaces written by third parties.

Reviewer: ```suggestion
The following section was written before WASI Preview 2 was stable. wasmCloud supports many [wasi-cloud-core](https://github.com/WebAssembly/wasi-cloud-core) interfaces, the existing `wasmcloud:*` interfaces, and custom interfaces written by third parties.
```

NIT: We don't support *all* of them just yet, right?

I'm fine with the current wording, but "many" might be a bit more accurate than "the". Or maybe just delete the "the".

:::

Capability contracts specify the required functionality for a given capability -- forming an interface or service abstraction that actors can rely on.

Capability contracts are identified with IDs -- unique names that identify an interface or service abstraction.

Capbility contract IDs in wasmCloud contain a vendor ID prefix (the vendor that defines the _contract_, not necessarily a specific _implementation_).

For example, the following is a list of some capability contract IDs supported by wasmCloud:

> - `wasmcloud:httpserver`

Reviewer: ```suggestion
- [`wasmcloud:httpserver`](https://github.com/wasmCloud/interfaces/tree/main/httpserver)
```

While we're here what do you think about making these links?

[host]: https://github.com/wasmCloud/wasmcloud

### Capability Providers

In wasmCloud, _capability providers_ are self-contained processes that implement a _capability contract_, which corresponds to a well-defined API.

Capability providers run on your wasmCloud lattice, and ensure that functionality depended on by actors is available.

### Deny-by-default Capability Access

Not every actor should be able to access every capability. Securing and restricting communication between actors and providers is important to avoid actors gaining access to capabilities they should not have access to.

To contact a particular capability provider, an actor must:
- be [signed](/docs/hosts/security) with a the appropriate capability contract ID, which happens automatically during the actor build process

> - be [linked at runtime](./runtime-linking.mdx) to the capability provider

Reviewer: Should this link be absolute from `/docs` ? Not sure if we have a preference on this, but making it consistent might help with search/replace

Reviewer 2: Docusaurus should catch it if it fails, but agreed on consistency. Absolute URLs with no file extension is preferred. The markdown language server understands file paths and should help with correcting links if anything is moved.

```suggestion
- be [linked at runtime](/docs/concepts/runtime-linking) to the capability provider
```
